upgrade.md

```diff
@@ -17,6 +17,146 @@ The `storagePath` method of the `Illuminate\Contracts\Foundation\Application` in
+
+`send`, `html`, `text` & `plain` don't return the number of recipients anymore but an instance of `Illuminate\Mail\SentMessage` which contains the `Symfony\Component\Mailer\SentMessage` instance
+
+##### Renamed Swift methods to Symfony
+
+All Swift methods were renamed to an equivalent with Symfony in the method name. For example:
+
+    // BEFORE
+    $this->withSwiftMessage(function ($message) {
+        $message->getHeaders()->addTextHeader(
+            'Custom-Header', 'Header Value'
+        );
+    });
+
+    // AFTER
+    use Symfony\Component\Mime\Email;
+
+    $this->withSymfonyMessage(function (Email $message) {
+        $message->getHeaders()->addTextHeader(
+            'Custom-Header', 'Header Value'
+        );
+    });
+
+Please also take a look at the [Symfony Mailer Docs](https://symfony.com/doc/6.0/mailer.html#creating-sending-messages) for all possible interactions with the Email message.
+
+Here is a list of all Methods which need to be renamed if they are used:
+    
+    Message::getSwiftMessage();
+    Message::getSymfonyMessage();
+
+    Mailable::withSwiftMessage($callback);
+    Mailable::withSymfonyMessage($callback);
+
+    MailMessage::withSwiftMessage($callback);
+    MailMessage::withSymfonyMessage($callback);
+
+    Mailer::getSwiftMailer();
+    Mailer::getSymfonyTransport();
+
+    Mailer::setSwiftMailer($swift);
+    Mailer::setSymfonyTransport(TransportInterface $transport);
+    
+    MailManager::createTransport($config);
+    MailManager::createSymfonyTransport($config);
+
+##### `Illuminate\Mail\Message`
+
+The `Illuminate\Mail\Message` class now contains an instance of `Symfony\Component\Mime\Email` instead of `Swift_Message` so all forwarding calls in user-land will need to be updated. Here are some examples which were possible with SwiftMailer but are not possible anymore with Symfony Mailer (most people are probably using the Laravel style already):
+
+    // IF YOU USED THIS
+    $message
+        ->setFrom('taylor@laravel.com')
+        ->setTo('example@example.org')
+        ->setSubject("Order Shipped")
+        ->setBody(
+            '<h1>HTML email</h1><p>Shows HTML tags if the client supports it.</p>',
+            'text/html'
+        )
+        ->addPart(
+            'Some Plain text in the email',
+            'text/plain'
+        );
+
+    // YOU NEED TO CHANGE IT TO
+    $email
+        ->from('taylor@laravel.com')
+        ->to('example@example.org')
+        ->subject("Order Shipped")
+        ->html('<h1>HTML email</h1><p>Shows HTML tags if the client supports it.</p>')
+        ->text('Some Plain text in the email');
+
+##### Removed `auth_mode` for SMTP
+
+Setting the `auth_mode` in the mail config was removed, because the authentication mode can be automatically negotiated between the Mailer and the SMTP server.
+
+##### Failed recipients
+
+It's no longer possible to get a list of failed recipients. Instead, if an email is failed to send, and exception will be thrown.
```

Comment:
What kind of exception?

Reply:
@taylorotwell 
A `TransportExceptionInterface` will be thrown if the email could not be send.
https://symfony.com/doc/current/mailer.html#handling-sending-failures

The failed recipients are an artifact of `Swift_RfcComplianceException`, thus holds the invalid address (you can validate this yourself as well, before sending)
https://github.com/symfony/symfony/issues/33812

It would probably be better to word it something like this:
```suggestion
It's no longer possible to get a list of failed recipients. Instead, you need to validate the emails yourself before sending.

Furthermore to validate if an email was really send you need to check this with your E-Mail Provider. For example through webhooks.
```